src/components/PostList.tsx

```diff
@@ -19,59 +21,73 @@ interface PostListProps {
   feedType: FeedTypes;
 }
 
-const sortFeed = (feed: FeedItem[]): FeedItem[] => {
-  feed.sort(
-    (firstFeedItem: FeedItem, secondFeedItem: FeedItem) =>
-      secondFeedItem.blockNumber - firstFeedItem.blockNumber
-  );
-  return feed;
-};
+const sortFeed = (feed: FeedItem[]): FeedItem[] =>
+  feed.sort(latestBatchedMessageComparator);
+
+const itemsPerPage = 5;
 
 const PostList = ({ feedType }: PostListProps): JSX.Element => {
+  const [currentFeed, setCurrentFeed] = React.useState<FeedItem[]>([]);
+  const [visibleItems, setVisibleItems] = React.useState<FeedItem[]>([]);
+
   const userId: string | undefined = useAppSelector((state) => state.user.id);
   const displayId: string | undefined = useAppSelector(
     (state) => state.user.displayId
   );
-  const myGraph: Record<string, RelationshipState> = useAppSelector(
-    (state) => (userId && state.graphs.following[userId]) || {}
+  const myGraph:
+    | Record<string, RelationshipState>
+    | undefined = useAppSelector((state) =>
+    userId ? state.graphs.following[userId] : undefined
   );
 
-  const initialFeed: FeedItem[] = useAppSelector(
-    (state) => state.feed.feedItems
-  ).filter(
-    (post) => post?.content?.type === "Note" && post?.inReplyTo === undefined
-  );
+  const feed = useAppSelector((state) => {
+    return state.feed.feedItems;
+  });
+
   const loading: boolean = useAppSelector(
     (state) => state.feed.isPostLoading.loading
   );
 
-  const feed: FeedItem[] = sortFeed(initialFeed);
+  useEffect(() => {
+    const feedTypeFilters = {
+      [FeedTypes.MY_FEED]: (p: FeedItem) =>
+        p.fromId === userId ||
+        (myGraph && myGraph[p.fromId]?.status === RelationshipStatus.FOLLOWING),
+      [FeedTypes.MY_POSTS]: (p: FeedItem) => userId === p?.fromId,
+      [FeedTypes.DISPLAY_ID_POSTS]: (p: FeedItem) => displayId === p?.fromId,
+      [FeedTypes.DISCOVER]: (_p: FeedItem) => true,
```

Comment:
maybe add this one in here are unknown ((_p) => true)

Reply:
I'd like to do that, but there isn't a FeedType.UNKNOWN enum value. The unexpected case is handled on line 61 with the `|| ((_p) => true)`